Userland/Libraries/LibCore/EventLoop.cpp

```diff
@@ -500,6 +506,16 @@ size_t EventLoop::pump(WaitMode mode)
         }
     }
 
+    // Strictly speaking, this is not necessary for proper functioning of automatic job cancellation.
+    // However, it (1) saves memory by removing finished jobs and (2) signals that some work was done.
+    m_running_jobs.remove_all_matching([&](auto const& job) {
```

Comment:
`EventLoop::pump` can become pretty hot (it is called at least for every mouse movement and repaint event), and this list can be arbitrarily long. This looks like it could end up using a lot of running time.

Suggestion: Measure how long `m_running_jobs` becomes in practice; I assume it's pretty low for now. Also, maybe add a warning if this mechanism is ever under heavy load? Something like:
```c++
if (m_running_jobs.size() > 30 && !s_warned_job_count) {
    s_warned_job_count = true;
    dbgln("WARNING: Job queue wasn't designed for this load ({} jobs). Please begin optimzing EventLoop::pump() -> m_running_jobs.remove_all_matching", m_running_jobs.size());
}
```